From 1cfd84e7effde04a8acb17c5b6333bccb18934d7 Mon Sep 17 00:00:00 2001
From: Dan O'Donovan <dan@emutex.com>
Date: Wed, 25 Mar 2015 17:41:08 +0000
Subject: [PATCH 258/389] tty: serial: sc16is7xx: Disable regcache when
 updating FCR register

The FCR register contains a write-only bit to reset the RX FIFO.
Without disabling the regmap, subsequent attempts to update this
bit may be silently supressed.  The cache should be disabled when
updating this bit.

Signed-off-by: Dan O'Donovan <dan@emutex.com>
Signed-off-by: Kartikey Parmar <kartikeyx.r.parmar@intel.com>
Change-Id: If669946ce7c7fe43f42458a70c7738a1dcef422e
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/4653
Reviewed-by: Fung, Joshua <joshua.fung@intel.com>
Tested-by: Fung, Joshua <joshua.fung@intel.com>
---
 drivers/tty/serial/sc16is7xx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/tty/serial/sc16is7xx.c b/drivers/tty/serial/sc16is7xx.c
index 0e791e9..a450851 100644
--- a/drivers/tty/serial/sc16is7xx.c
+++ b/drivers/tty/serial/sc16is7xx.c
@@ -528,9 +528,11 @@ static void sc16is7xx_handle_rx(struct uart_port *port, unsigned int rxlen,
 		if (unlikely(lsr & SC16IS7XX_LSR_OE_BIT)) {
 			/* Reset Rx FIFO to clear overrun status */
 			dev_warn(port->dev, "Resetting RX FIFO due to overrun");
+			regcache_cache_bypass(s->regmap, true);
 			sc16is7xx_port_update(port, SC16IS7XX_FCR_REG,
 					      SC16IS7XX_FCR_RXRESET_BIT,
 					      SC16IS7XX_FCR_RXRESET_BIT);
+			regcache_cache_bypass(s->regmap, false);
 			break;
 		}
 	}
-- 
1.9.3

